pkg/prowloader/prow.go

```diff
@@ -268,21 +269,23 @@ func (pl *ProwLoader) extractTestCases(suite *junit.TestSuite, testCases map[str
 			continue
 		}
 
-		status := 12
+		status := v1.TestStatusFailure
 		if tc.FailureOutput == nil {
-			status = 1
+			status = v1.TestStatusSuccess
 		}
 
 		key := fmt.Sprintf("%s.%s", suite.Name, tc.Name)
 		if existing, ok := testCases[key]; !ok {
 			testCases[key] = &models.ProwJobRunTest{
 				TestID:   pl.findOrAddTest(tc.Name),
 				SuiteID:  pl.findOrAddSuite(suite.Name),
-				Status:   status,
+				Status:   int(status),
 				Duration: tc.Duration,
 			}
-		} else {
-			existing.Status += status
+		} else if (existing.Status == int(v1.TestStatusFailure) && status == v1.TestStatusSuccess) ||
+			(existing.Status == int(v1.TestStatusSuccess) && status == v1.TestStatusFailure) {
+			// One pass among failures makes this a flake
+			existing.Status = int(v1.TestStatusFlake)
```

Comment:
Actually I think this isn't correct, because I need to consider the suite, too

Reply:
Ah past me did consider the suite, the map key prefixes the suite name.  So this should be correct.